rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,571 @@
+- `nextTripKey` (Trip Key, optional): links the newly created trip to the trip after it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message. Required if `startLocation` is specified and `departureTime` is NOT specified.
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+- `comment` (string, optional): free text with additional information about the trip
+
+*Notes*
+- At least one of `startLocation` and `endLocation` is required.
+- If `startLocation` is 
+- The event should only include values which were explicitly included by the `author`: inferred values should not be included.
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip dropped
+This removes a trip (either added or scheduled), indicating that it will not be running.
+
+Event type: `com.mbta.ctd.glides.trip-dropped`
+Fields in the event:
+- `author` (Author): the inspector dropping the trip.
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+- `reason` (string): a text description of why the trip is being dropped.
+
+### Trip restored
+This restores a trip (either added or scheduled) that was previously dropped.
+
+Event type: `com.mbta.ctd.glides.trip-restored`
+Fields in the event:
+- `author` (Author): the inspector restoring the trip.
+- `tripKey` (Trip Key): which trip to restore. can be either an added or scheduled trip.
+
+### Trip updated
+This indicates that a trip was updated in some fashion:  consist, operators, departure time. This includes updates which confirm an arrival or departure time. Fields which are not present are not considered to be updated.
+
+Event type: `com.mbta.ctd.glides.trip-updated`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): which trip is being updated
+- `comment` (string, optional): free text information about the trip
+- `route` (string, optional): if specified, the new route tag for the trip. This is different from a GTFS route ID.
+- `location` (Location, optional): where the time/consist is being updated for. This can be different from the location in the `author` value or in the `tripKey`, if the author is updating information about the train relative to a different location.
```

Comment:
Updating information about the train relative to a location you aren't editing for is not currently supported in Glides and we don't expect it to be soon, so it's out of scope and shouldn't be here.

Reply:
In https://github.com/mbta/technology-docs/pull/18#discussion_r1084404009 @skyqrose mentioned the possibility of dispatchers adding trips from other locations. While not in the current Glides functionality, "some additional flexibility is included".

Reply:
I agree this is fine. If we want events to be relative to a location like they're entered in Glides, it'd be nice if the event data spoke for itself without needing to reference the author for context (unless you actually care about the author).